pkg/agent/monitortool/latency_store_test.go

```diff
@@ -0,0 +1,245 @@
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			entry, _ := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry.LastMeasuredRTT, entry.LastMeasuredRTT)
+			assert.Equal(t, tt.expectedEntry.LastSendTime, entry.LastSendTime)
+			assert.Equal(t, tt.expectedEntry.LastRecvTime, entry.LastRecvTime)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key               string
+		prevExpectedEntry NodeIPLatencyEntry
+		expectedEntry     NodeIPLatencyEntry
+	}{
+		{
+			key:               "10.244.2.1",
+			prevExpectedEntry: entry,
+			expectedEntry:     NodeIPLatencyEntry{},
+		},
+		{
+			key:               "10.244.2.2",
+			prevExpectedEntry: NodeIPLatencyEntry{},
+			expectedEntry:     NodeIPLatencyEntry{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			entry, _ := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.prevExpectedEntry, entry)
+			latencyStore.DeleteNodeIPLatencyEntry(tt.key)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, entry, tt.expectedEntry)
+			assert.False(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_SetNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry2,
+			expectedEntry: entry2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = tt.updatedEntry.LastSendTime
+				entry.LastRecvTime = tt.updatedEntry.LastRecvTime
+				entry.LastMeasuredRTT = tt.updatedEntry.LastMeasuredRTT
+			}
+			latencyStore.SetNodeIPLatencyEntry(tt.key, mutator)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry, entry)
+			assert.True(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteStaleNodeIPs(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
there is no reason to use table-driven tests if you have a single test case, it just hurts readability

Reply:
Thanks, I have changed it to one check.